### PR TITLE
Change the implementaion to use the bedrock client library

### PR DESCRIPTION
**Pull request by Mariappan**
[[BDRK-1093]](https://basis-ai.atlassian.net/browse/BDRK-1093) - Update download_artefact code snippet to use client library

Change the implementation of download_artefact using client library

**Tests:**
* Tested download_and_unzip_latest_artefact api for download_artefacts.
* The other api download_artefact_by_run_id() has following issue:

**Note:** pipeline_api.get_training_pipeline_run api currently returning None for artefact_id. 
Need to fix it. Creating PR for getting overall comments.

[BDRK-1093]: https://basis-ai.atlassian.net/browse/BDRK-1093

### CODE REVIEW

File: utils/artefact.py [hunk 1]
```diff
@@ -1,93 +1,98 @@
...
 import os
-import zipfile
+
 from typing import Optional
+from bdrk.v1_util import download_and_unzip_artefact, get_artefact_stream
+from bdrk.v1 import ApiClient, Configuration, ModelApi, PipelineApi, ServeApi
+
+configuration = Configuration()
+
+configuration.api_key["X-Bedrock-Access-Token"] = os.environ["BEDROCK_ACCESS_TOKEN"]
```
Comment (nglinh) [inline utils/artefact.py]:
```suggestion
# This is your personal access token https://docs.basis-ai.com/getting-started/rest-apis/personal-api-token
configuration.api_key["X-Bedrock-Access-Token"] = os.environ["BEDROCK_ACCESS_TOKEN"]
```
Let's add a link to the doc

File: utils/artefact.py [hunk 2]
```diff
@@ -1,93 +1,98 @@
...
+configuration.host = os.environ['BEDROCK_API_DOMAIN']
```
Comment (nglinh) [inline utils/artefact.py]: let's just put  this to `api.bdrk.ai`, no need to make it too configurable

File: utils/artefact.py [hunk 3]
```diff
@@ -1,93 +1,98 @@
...
-        ) from exc
+    last_run = _get_latest_run(pipeline_id)
 
     # Calls Bedrock API again to download artefact from the latest run
-    filename = download_artefact_by_run_id(
-        pipeline_run_id=last_run["entity_id"], output_filepath=output_filepath
-    )
+    filename = download_artefact_by_run_id(pipeline_id=pipeline_id,
+                                           pipeline_run_id=last_run.entity_id,
+                                           output_filepath=output_filepath)
```
Comment (sweatybridge) [inline utils/artefact.py]: [nitpick] let's be consistent with indentations.
```python
    filename = download_artefact_by_run_id(
        pipeline_id=pipeline_id,
        pipeline_run_id=last_run.entity_id,
        output_filepath=output_filepath
    )
```

File: utils/artefact.py [hunk 4]
```diff
@@ -113,13 +118,14 @@ def download_and_unzip_artefact(output_directory: Optional[str] = None) -> bool:
...
+    last_run = _get_latest_run(pipeline_id)
 
     # Extract the downloaded artefacts
-    output_directory = output_directory or f"/tmp/{pipeline_id}"
-    with zipfile.ZipFile(filename, "r") as f:
-        f.extractall(path=output_directory)
+    download_and_unzip_artefact(api_client=api_client,
+                                model_id=pipeline.model_id,
+                                model_artefact_id=last_run.artefact_id,
+                                output_dir=output_directory or f"/tmp/{pipeline_id}")
```
Comment (sweatybridge) [inline utils/artefact.py]: ditto

File: utils/artefact.py (hunk 2 above)
Comment (sweatybridge) [reply utils/artefact.py]: We need this configuration to run this code snippet in our staging environments.

Comment (nglinh) [reply utils/artefact.py]: I see. then we should have the default to api.bdrk.ai while allowing overrides

**Comment by Mariappan** [comment]
span-api bug is fixed inthe following PR:
https://github.com/basisai/span/pull/1204

**Tests**
``` python
>>> os.environ["BEDROCK_API_DOMAIN"] = "http://localhost:8000"
>>> import utils.artefact as arte
>>> arte.download_and_unzip_latest_artefact("/tmp/api3")
['/tmp/api3/mobydick.txt', '/tmp/api3/kueh lapis']
True
>>> os.listdir("/tmp/api3")
['mobydick.txt', 'kueh lapis']
>>> arte.download_artefact_from_latest_run(pipeline_id='firstpipeline-396d39ad', output_filepath="/tmp/api2")
Downloading artefact for pipeline firstpipeline-396d39ad and run: cee09c0b-fb4f-4143-93a3-9984bf6fb4bb
Downloaded artefact: /tmp/api2
'/tmp/api2'
>>> arte.download_artefact_from_latest_run(pipeline_id='firstpipeline-396d39ad')
Downloading artefact for pipeline firstpipeline-396d39ad and run: cee09c0b-fb4f-4143-93a3-9984bf6fb4bb
Downloaded artefact: /tmp/cee09c0b-fb4f-4143-93a3-9984bf6fb4bb-artefact.zip
'/tmp/cee09c0b-fb4f-4143-93a3-9984bf6fb4bb-artefact.zip'
>>> arte.download_artefact_by_run_id('firstpipeline-396d39ad', '1894d690-8dcb-4811-affd-fa57f74c55f0')
Downloading artefact for pipeline firstpipeline-396d39ad and run: 1894d690-8dcb-4811-affd-fa57f74c55f0
'/tmp/1894d690-8dcb-4811-affd-fa57f74c55f0-artefact.zip'

```
``` bash
maari@maarvis:/tmp$ file api2
api2: Zip archive data, at least v2.0 to extract
maari@maarvis:/tmp$ ls *artefact.zip
cee09c0b-fb4f-4143-93a3-9984bf6fb4bb-artefact.zip
maari@maarvis:/tmp$ 
maari@maarvis:/tmp$ ls *artefact.zip
1894d690-8dcb-4811-affd-fa57f74c55f0-artefact.zip  cee09c0b-fb4f-4143-93a3-9984bf6fb4bb-artefact.zip
maari@maarvis:/tmp$ tree api3
api3
├── kueh lapis
└── mobydick.txt

0 directories, 2 files
maari@maarvis:/tmp$ 

```

**Comment by nglinh** [comment]
This depends on the new release of the client library also. Please release first before merging this in

**Comment by Mariappan** [comment]
Python package released:
https://pypi.org/project/bdrk/0.1.4/

**Pull request merged**